### ISSUE TITLE
IDE standardization?

### ISSUE THREAD

**Issue by NoraCodes** [ipfs/py-ipfs#7]
Should we include .idea/ for PyCharm? I use PyCharm extensively on my desktop dev machine, and I think it would make the process of onboarding new developers easier if we had a .idea/ project directory already set up with execution settings for tests, etc. already set up. Otherwise, we will end up with a lot of duplication of work.

**Comment by BrendanBenshoof** [comment]
So long as IDE integration is optional to the build and testing process.
(We should always be able to do everything needed from the command line)

**Comment by bmcorser** [comment]
Agreed. No harm in offering a PyCharm config, though. Many people swear by
it `;)`

**Comment by NoraCodes** [comment]
@BrendanBenshoof Absolutely. I do all my development from CLI on my laptop, so I'll need that myself.

**Comment by JulienPalard** [comment]
Not sure about your PyCharm thing, I don't know what the `.idea` contains. But just keep it optional, let everyone use the IDE / editor they want, I propose our only common standardisation point to be the PEP8, and it should be "enough" standard ?

**Comment by NoraCodes** [comment]
Fair enough. `.idea` basically just contains configs for running tests from within the IDE, which is nice for those who don't like to have more than their IDE open. I personally would be OK with not including them, since as @BrendanBenshoof says, testing should always work without extra tools.

**Comment by JulienPalard** [comment]
Still don't exactly know what is inside the `.idea`, but, if it helps those with a set of IDE without impairing others, I'm not 100% against it, so, same as @BrendanBenshoof here.

**Comment by mvanveen** [comment]
I live in vi and I imagine many others do, but if the `.idea` isn't large let's go ahead and add it.  Alternatively, we could keep an ipfs URI for it around :)

**Comment by NoraCodes** [comment]
@mvanveen  That's a much better idea, actually. I'd be happy to go with that.
